connectors/services/job_execution.py

```diff
@@ -46,11 +47,15 @@ async def _sync(self, sync_job):
             logger.error(f"Couldn't find connector by id {sync_job.connector_id}")
             return
 
-        if connector.last_sync_status == JobStatus.IN_PROGRESS:
-            logger.debug(
-                f"Connector {connector.id} is still syncing, skip the job {sync_job.id}..."
-            )
-            return
+        if (
+            sync_job.job_type == JobType.FULL
+            or sync_job.job_type == JobType.INCREMENTAL
+        ):
+            if connector.last_sync_status == JobStatus.IN_PROGRESS:
```

Comment:
```suggestion
        supported_job_types = [JobType.FULL, JobType.INCREMENTAL]

        if (sync_job.job_type in supported_job_types 
        and connector.last_sync_status == JobStatus.IN_PROGRESS):
```

nit: I think this could improve readability (removing one level of nesting)

Reply:
I'm thinking you will add the access control check here like

```python
if (
    sync_job.job_type == JobType.FULL
    or sync_job.job_type == JobType.INCREMENTAL
):
    if connector.last_sync_status == JobStatus.IN_PROGRESS:
        logger.debug(
            f"Connector {connector.id} is still syncing, skip the job {sync_job.id}..."
        )
        return
elif sync_job.job_type == JobType.ACCESS_CONTROL:
    if connector.last_access_control_sync_status == JobStatus.IN_PROGRESS:
        logger.debug(...)
        return
```